client/state/checklist/reducer.js

```diff
@@ -14,15 +14,25 @@ const markChecklistTaskComplete = ( state, taskId ) => ( {
 	tasks: { ...state.tasks, [ taskId ]: true },
```

Comment:
Whoops, this has the wrong shape:
```suggestion
	tasks: { ...state.tasks, [ taskId ]: { completed: true } },
```

Not sure whether we ever expect other properties to go in there or we can make a new object, but we might as well fix this up here. It's not behaving as it should right now.

Reply:
> Whoops, this has the wrong shape: [...]

Right :thinking: https://github.com/Automattic/wp-calypso/blob/a2220b019a5b4d91aa30d22bad3895b875ed644c/client/my-sites/plans/current-plan/jetpack-checklist/index.tsx#L64-L66

Hmm seems like it has been like this for a while :slightly_frowning_face: https://github.com/Automattic/wp-calypso/pull/33193/files#diff-0c8c68f71be23c8ded46a59c076439c1L34